avocado/plugins/expected_files_merge.py

```diff
@@ -0,0 +1,149 @@
+
+"""
+Functions for merging equal expected files together
+"""
+
+import os
+import shutil
+
+from avocado.core.plugin_interfaces import JobPost, CLI
+from avocado.utils import genio
+
+
+def _merge_expected_files(parent_directory):
+    """
+    Finds all output, stdout and stderr expected files in subdirectories and
+    groups equal files together. After that finds the biggest group for every
+    file type and creates file in parent directory which is equal to all files
+    in group. At the end it deletes files in group because they are no long needed.
+    :param parent_directory: Directory where should be merge file created
+    :type parent_directory: str
+    """
+
+    stdout_dict = {}
+    stderr_dict = {}
+    output_dict = {}
+    for path in os.listdir(parent_directory):
+        variants_level_path = os.path.join(parent_directory, path)
+        if os.path.isdir(variants_level_path):
+            for file in os.listdir(variants_level_path):
+                file = os.path.join(variants_level_path, file)
+                if file.endswith('output.expected'):
+                    _save_file_to_group(file, output_dict)
+                elif file.endswith('stdout.expected'):
+                    _save_file_to_group(file, stdout_dict)
+                elif file.endswith('stderr.expected'):
+                    _save_file_to_group(file, stderr_dict)
+    if output_dict:
```

Comment:
What about something like:

```python
    for file_dict, filename in ((output_dict, 'output.expected'),
                                (stdout_dict, 'stdout.expected'),
                                (stderr_dict, 'stderr.expected')):
        if file_dict:
            merged_file = _get_best_group(file_dict)
            if merged_file is not None:
                not_useful_files = file_dict[merged_file]
                shutil.copyfile(merged_file, os.path.join(parent_directory,
                                                          filename))
                _delete_files(not_useful_files)
```

Instead of repeating the same code?

Reply:
Yes, this is much better, thanks